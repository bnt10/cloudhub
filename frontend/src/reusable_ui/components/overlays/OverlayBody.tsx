import React, {FunctionComponent, ReactNode} from 'react'

interface Props {
  children: ReactNode
}

const OverlayBody: FunctionComponent<Props> = ({children}) => (
  <div className="overlay--body">{children}</div>
)

export default OverlayBody
