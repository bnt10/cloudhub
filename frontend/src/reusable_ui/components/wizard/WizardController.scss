/*
  Wizard Controller Styles
  ------------------------------------------------------------------------------
*/

.wizard-step-title {
  text-align: center;
  @include no-user-select();
}

.wizard-controller {
  display: inline-flex;
  flex-direction: column;
  min-width: 100%;
  align-items: center;
}

.wizard-skip-link {
  flex-shrink: 0;
}

.wizard-tip-text {
  min-width: 100%;
  color: $g12-forge;
  text-align: center;
  p {
    font-weight: 600;
  }
}
