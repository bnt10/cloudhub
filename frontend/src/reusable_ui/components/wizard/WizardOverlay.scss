/*
  Overlay Wizard Styles
  ------------------------------------------------------------------------------
*/

.wizard-overlay {
  .overlay--body {
    padding: 30px 0;
  }
}