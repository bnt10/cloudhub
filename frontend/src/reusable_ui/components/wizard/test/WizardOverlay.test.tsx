import React from 'react'
import {shallow} from 'enzyme'

import WizardOverlay from 'src/reusable_ui/components/wizard/WizardOverlay'
import WizardController from 'src/reusable_ui/components/wizard/WizardController'
import OverlayBody from 'src/reusable_ui/components/overlays/OverlayBody'
import OverlayContainer from 'src/reusable_ui/components/overlays/OverlayContainer'
import OverlayTechnology from 'src/reusable_ui/components/overlays/OverlayTechnology'
import OverlayHeading from 'src/reusable_ui/components/overlays/OverlayHeading'

describe('WizardOverlay', () => {
  let wrapper

  const wrapperSetup = (override = {}) => {
    const props = {
      children: null,
      visible: undefined,
      title: undefined,
      toggleVisibility: () => jest.fn(),
      resetWizardState: jest.fn(),
      skipLinkText: undefined,
      maxWidth: undefined,
      jumpStep: undefined,
      ...override,
    }

    return shallow(<WizardOverlay {...props} />)
  }

  beforeEach(() => {
    jest.resetAllMocks()
    wrapper = wrapperSetup()
  })

  it('mounts without exploding', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('renders no WizardController component', () => {
    expect(wrapper.find(WizardController)).toHaveLength(0)
  })

  it('renders no OverlayTechnology component', () => {
    expect(wrapper.find(OverlayTechnology)).toHaveLength(1)
  })

  it('renders no OverlayContainer component', () => {
    expect(wrapper.find(OverlayContainer)).toHaveLength(1)
  })

  it('renders no OverlayHeading component', () => {
    expect(wrapper.find(OverlayHeading)).toHaveLength(1)
  })

  it('renders no OverlayBody component', () => {
    expect(wrapper.find(OverlayBody)).toHaveLength(1)
  })

  it('matches snapshot with minimal props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('with children', () => {
    it('renders one WizardController component', () => {
      wrapper = wrapperSetup({children: {}})
      expect(wrapper.find(WizardController)).toHaveLength(1)
    })

    it('matches snapshot with children props', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
