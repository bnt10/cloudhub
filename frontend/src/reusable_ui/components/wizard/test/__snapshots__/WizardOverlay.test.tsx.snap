// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardOverlay matches snapshot with minimal props 1`] = `
<div
  className="wizard-overlay"
>
  <OverlayTechnology>
    <OverlayContainer
      maxWidth={800}
    >
      <OverlayHeading
        onDismiss={[Function]}
      />
      <OverlayBody />
    </OverlayContainer>
  </OverlayTechnology>
</div>
`;

exports[`WizardOverlay with children matches snapshot with children props 1`] = `
<div
  className="wizard-overlay"
>
  <OverlayTechnology>
    <OverlayContainer
      maxWidth={800}
    >
      <OverlayHeading
        onDismiss={[Function]}
      />
      <OverlayBody />
    </OverlayContainer>
  </OverlayTechnology>
</div>
`;
